test/common/config/subscription_factory_impl_test.cc

```diff
@@ -69,14 +71,33 @@ class SubscriptionFactoryTest : public testing::Test {
   NiceMock<LocalInfo::MockLocalInfo> local_info_;
   NiceMock<ProtobufMessage::MockValidationVisitor> validation_visitor_;
   Api::ApiPtr api_;
-  NiceMock<Runtime::MockLoader> runtime_;
   SubscriptionFactoryImpl subscription_factory_;
 };
 
+class SubscriptionFactoryTestUnifiedOrLegacyMux
+    : public SubscriptionFactoryTest,
+      public testing::WithParamInterface<LegacyOrUnified> {
+public:
+  SubscriptionFactoryTestUnifiedOrLegacyMux()
+      : should_use_unified_(GetParam() == LegacyOrUnified::Unified) {
+    if (should_use_unified_) {
```

Comment:
I don't think `should_use_unified_` is used anywhere else in the class, so the data member is not really necessary here.